### PR TITLE
Implement xbmcgui.Dialog().browse()

### PR DESCRIPTION
**Pull request by michaelarnauts**
### Functional description

* Implement `xbmcgui.Dialog().browse()`, `xbmcgui.Dialog().browseSingle()` and `xbmcgui.Dialog().browseMultiple()`
* Correct typo
* Fix interactive usage of `xbmcgui.Dialog().numeric()` and `xbmcgui.Dialog().input()`

### CODE REVIEW

File: xbmcgui.py [hunk 1]
```diff
@@ -262,11 +261,93 @@ def numeric(self, type, heading, defaultt=None, bHiddenInput=None):
...
+            return ''
+
+    def browseMultiple(self, type, heading, shares, mask='', useThumbs=False, treatAsFolder=False, defaultt=''):
+        """ Browser dialog for multiple files.
+
+        See xbmcgui.browse.
+
+        :return: Returns tuple of marked filenames as a string, if user pressed 'Ok' or a masked item was selected.
+                 Returns empty tuple if dialog was canceled.
+        :rtype: str
```
Comment (basrieter) [inline xbmcgui.py]: The return type should be a tuple of string?

File: xbmcgui.py [hunk 2]
```diff
@@ -262,11 +261,93 @@ def numeric(self, type, heading, defaultt=None, bHiddenInput=None):
...
+        KodiStub.print_heading(heading)
+        try:
+            answer = self.read_input("Please enter the path to a file [{}]?".format(defaultt), color=Colors.Yellow)
+            return answer if answer is not '' else defaultt
+        except EOFError:
+            return ''
+
+    def browseMultiple(self, type, heading, shares, mask='', useThumbs=False, treatAsFolder=False, defaultt=''):
+        """ Browser dialog for multiple files.
+
```
Comment (basrieter) [inline xbmcgui.py:374]: Could you please provide the correct parameter documentation in reStructuredText style? This helps debugging and type checking.

File: xbmcgui.py [hunk 3]
```diff
@@ -262,11 +261,93 @@ def numeric(self, type, heading, defaultt=None, bHiddenInput=None):
...
+        :return: Returns tuple of marked filenames as a string, if user pressed 'Ok' or a masked item was selected.
+                 Returns empty tuple if dialog was canceled.
+        :rtype: str
+
+        """
+        return self.browse(type, heading, shares, mask, useThumbs, treatAsFolder, defaultt, True)
+
+    def browseSingle(self, type, heading, shares, mask='', useThumbs=False, treatAsFolder=False, defaultt=''):
+        """ Browser dialog for single files.
+
```
Comment (basrieter) [inline xbmcgui.py:398]: Could you please provide the correct parameter documentation in reStructuredText style? This helps debugging and type checking.

**Review by basrieter — CHANGES_REQUESTED** [review]
Thanks! Would it possible to complete the parameter documentation in reStructuredText style?

File: xbmcgui.py (hunk 2 above)
Comment (michaelarnauts) [reply xbmcgui.py:374]: Sure, do you want to repeat the same block from above then? I've looked for something like inheritDoc that exists in phpdoc, bit could not find it.

Comment (basrieter) [reply xbmcgui.py:374]: Yeh, no such thing in Python, so just repeat them.

**Review by basrieter — APPROVED** [review]

**Comment by basrieter** [comment]
Thanks!

**Pull request merged**